### PR TITLE
Allow superpmi.exe to find a default JIT

### PR DESCRIPTION
**Pull request by BruceForstall**
This makes it slightly easier to invoke superpmi.exe manually: most of the time you want to use the clrjit.dll (or similar) in the same directory where superpmi.exe lives (in a Core_Root directory), so use that if not JIT is specified.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@jakobbotsch PTAL
cc @dotnet/jit-contrib

File: src/coreclr/tools/superpmi/superpmi/commandline.cpp [hunk 1]
```diff
@@ -669,12 +659,164 @@ bool CommandLine::Parse(int argc, char* argv[], /* OUT */ Options* o)
             return false;
         }
     }
+
+    SPMI_TARGET_ARCHITECTURE DefaultSpmiTargetArchitecture = GetSpmiTargetArchitecture();
```
Comment (jakobbotsch) [inline src/coreclr/tools/superpmi/superpmi/commandline.cpp:663]: Wrong naming conventions (here and below)

File: src/coreclr/tools/superpmi/superpmi/commandline.cpp [hunk 2]
```diff
@@ -669,12 +659,164 @@ bool CommandLine::Parse(int argc, char* argv[], /* OUT */ Options* o)
...
+#if defined(HOST_UNIX)
+        host_os_tag = "unix";
+        jit_host_os_prefix = "lib";
+        jit_host_os_extension = ".so";
+#elif defined(HOST_WINDOWS)
+        host_os_tag = "win";
+        jit_host_os_extension = ".dll";
+#else
+        // TODO: handle Mac
+        allowDefaultJIT = false;
```
Comment (jakobbotsch) [inline src/coreclr/tools/superpmi/superpmi/commandline.cpp:692]: Needs to be addressed? Presumably it's just .dylib?

**Review by jakobbotsch — COMMENTED** [review]
Seems fine to me, but looks like there are build failures.

File: src/coreclr/tools/superpmi/superpmi/commandline.cpp [hunk 3]
```diff
@@ -669,12 +659,170 @@ bool CommandLine::Parse(int argc, char* argv[], /* OUT */ Options* o)
...
+
+    if (o->nameOfJit == nullptr)
+    {
+        // Try to find a JIT in the same directory as superpmi.exe. If found (based on targetArchitecture), use it.
+
+        bool allowDefaultJIT = true;
+
+        const char* host_os_tag = "";
+        const char* jit_host_os_prefix = "";
+        const char* jit_host_os_extension = "";
```
Comment (jakobbotsch) [inline src/coreclr/tools/superpmi/superpmi/commandline.cpp:681]: Wrong naming convention

File: src/coreclr/tools/superpmi/superpmi/commandline.cpp [hunk 4]
```diff
@@ -669,12 +659,170 @@ bool CommandLine::Parse(int argc, char* argv[], /* OUT */ Options* o)
...
+                sprintf_s(tempStr, len, "%s%c%s%s_%s_%s_%s%s",
+                    programPath,
+                    DIRECTORY_SEPARATOR_CHAR_A,
+                    jitHostOSPrefix,
+                    jitBaseName,
+                    jitOSName,
+                    targetArch,
+                    hostArch,
+                    jitHostOSExtension);
+            }
```
Comment (jakobbotsch) [inline src/coreclr/tools/superpmi/superpmi/commandline.cpp:798]: Is there any reason to not just use the "default" JIT (clrjit.dll/libclrjit.so/libclrjit.dylib)?

**Review by jakobbotsch — APPROVED** [review]

File: src/coreclr/tools/superpmi/superpmi/commandline.cpp (hunk 4 above)
Comment (BruceForstall) [reply src/coreclr/tools/superpmi/superpmi/commandline.cpp:798]: We actually do use the default. Only if you use the `-target` option do the non-default ones get used.

**Pull request merged**